File: 프로그래머스/unrated/120808. 분수의 덧셈/분수의 덧셈.java
class Solution {
    public int[] solution(int numer1, int denom1, int numer2, int denom2) {
        int molecule = (numer1*denom2 + numer2*denom1);
        int denominator = denom1 * denom2;
        int gcd = findGCD(molecule, denominator);
        int[] answer = {molecule / gcd, denominator / gcd};
    return answer;
}

private int findGCD(int a, int b) {
    // 최대공약수 구하는 공식
    while (b != 0) {
        int temp = b;
        b = a % b;
        a = temp;
    }
    return a;
    }
}

File: 프로그래머스/0/120808. 분수의 덧셈/분수의 덧셈.py
def solution(numer1, denom1, numer2, denom2):
    numer3 = (numer1*denom2 + numer2*denom1)
    denom3 = denom1*denom2
    
    num = 1
    for i in range(2,min(numer3,denom3)+1):
        if numer3 % i == 0 and denom3 % i == 0:
            num = i
    
    return [numer3//num ,denom3//num ]